collated/main/A0139923X.md

```diff
@@ -0,0 +1,727 @@
+                            dataType.get().trim()
+                            );
+                } catch (IllegalValueException ive) {
+                    return new IncorrectCommand(ive.getMessage());
+                }
+            } else if (matcher_event.matches()) {
+                try {
+                    return new EditCommand(
+                            matcher_event.group("name").trim(), 
+                            matcher_event.group("date").trim(),
+                            isInputPresent(matcher_event.group("endDate")).trim(),
+                            matcher_event.group("startTime").trim(), 
+                            matcher_event.group("endTime").trim(),
+                            Integer.parseInt(matcher_event.group("targetIndex")), 
+                            dataType.get().trim()
+                            );
+                } catch (IllegalValueException ive) {
+                    return new IncorrectCommand(ive.getMessage());
+                }
+            } else if (matcher_deadline.matches()) {
+                try {
+                    return new EditCommand(
+                            matcher_deadline.group("name").trim(), 
+                            matcher_deadline.group("date").trim(),
+                            matcher_deadline.group("endTime").trim(), 
+                            Integer.parseInt(matcher_deadline.group("targetIndex")),
+                            dataType.get().trim()
+                            );
+                } catch (IllegalValueException ive) {
+                    return new IncorrectCommand(ive.getMessage());
+                }
+            } else {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+            }
+        } else {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+    }
+    
+    /*
+     *  Check whether the attribute is set
+     */
+```
+###### /java/seedu/todoList/logic/parser/Parser.java
+``` java
+    private String isInputPresent(String input){
+        return input == null ? "No End Date" : input;
+    }
+}
+```
+###### /java/seedu/todoList/model/ModelManager.java
+``` java
+    public synchronized void editTask(ReadOnlyTask target, String dataType, Task task) throws IllegalValueException, TaskNotFoundException {
+    	
+        /*
+         *  Scenario: User wants to edit todo to change to event or deadline
+         *  User enters edit todo 1 n/test d/01-01-2016 e/1000
+         *  This will remove index 1 of todolist and add into deadlinelist
+         *  
+         *  During edit process, example: edit todo 1 with parameters of deadline
+         *  todolist will remove the task index 1 and deadlinelist will add the edited task
+         */
+    	if(task instanceof Todo) {
+    	    if(dataType.equals("todo")){
+                todoList.addTask(task);
```

Comment:
you should try to do "in-place" edit
editing by adding and removing has its own set of problems that you can identify when you dog food the product